apiserver/storageprovisioner/internal/filesystemwatcher/watchers.go

```diff
@@ -0,0 +1,453 @@
+package filesystemwatcher
+
+import (
+	"gopkg.in/juju/names.v2"
+	"gopkg.in/tomb.v1"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/watcher"
+	"github.com/juju/utils/set"
+)
+
+// Watchers provides methods for watching filesystems. The watches aggregate
+// results from machine- and model-scoped watchers, to conform to the behaviour
+// of the storageprovisioner worker. The model-level storageprovisioner watches
+// model-scoped filesystems that have no backing volume. The machine-level worker
+// watches both machine-scoped filesytems, and model-scoped filesystems attached
+// to the machine.
+type Watchers struct {
+	Backend Backend
+}
+
+// WatchModelFilesystems returns a strings watcher that reports model-scoped
+// filesystems that have no backing volume. Volume-backed filesystems are
+// always managed by the machine to which they are attached.
+func (fw Watchers) WatchModelFilesystems() state.StringsWatcher {
```

Comment:
I'll change it to "ModelManaged" and "MachineManaged", to make it clear that we're interested in machines that are managed by the model-level or machine-level storage provisioner.